pkg/platform/kube/functionres/lazy.go

```diff
@@ -732,6 +763,84 @@ func (lc *lazyClient) getFunctionPods(ctx context.Context,
 	}
 }
 
+// getFunctionDeploymentOrPodLastWarningEvent returns the latest warning event message
+// for the function's associated Deployment object, or its pods if Deployment has none.
+func (lc *lazyClient) getFunctionDeploymentOrPodLastWarningEvent(ctx context.Context, function *nuclioio.NuclioFunction) string {
+	deploymentName := kube.DeploymentNameFromFunctionName(function.Name)
+
+	// Try to get Deployment warning first
+	warning := lc.getLastWarningEventForKindAndName(ctx, function.Namespace, "Deployment", deploymentName)
+	if warning != "" {
+		return warning
+	}
+
+	// If no deployment warning, check pods
+	if podLastEvent := lc.getLastWarningEventForFunctionPods(ctx, function); podLastEvent != "" {
+		return "Last warning event from one of the function's pods: " + podLastEvent
```

Comment:
I'd remove this message, as it makes the error a bit convoluted (as in the example attached to the description).